xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-ui/src/test/java/org/xwiki/refactoring/RefactoringConfigurationTest.java

```diff
@@ -0,0 +1,96 @@
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.refactoring;
+
+import org.jsoup.Jsoup;
+import org.junit.jupiter.api.Test;
+import org.xwiki.environment.Environment;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.rendering.internal.macro.html.renderers.html5.HTMLMacroHTML5Renderer;
+import org.xwiki.rendering.internal.renderer.html5.HTML5BlockRenderer;
+import org.xwiki.rendering.internal.renderer.html5.HTML5Renderer;
+import org.xwiki.rendering.internal.renderer.html5.HTML5RendererFactory;
+import org.xwiki.rendering.internal.renderer.xhtml.image.DefaultXHTMLImageRenderer;
+import org.xwiki.rendering.internal.renderer.xhtml.image.DefaultXHTMLImageTypeRenderer;
+import org.xwiki.rendering.internal.renderer.xhtml.link.DefaultXHTMLLinkRenderer;
+import org.xwiki.rendering.internal.renderer.xhtml.link.DefaultXHTMLLinkTypeRenderer;
+import org.xwiki.rendering.syntax.Syntax;
+import org.xwiki.test.annotation.ComponentList;
+import org.xwiki.test.junit5.mockito.MockComponent;
+import org.xwiki.test.page.PageTest;
+import org.xwiki.test.page.XWikiSyntax21ComponentList;
+import org.xwiki.velocity.tools.EscapeTool;
+import org.xwiki.xml.internal.html.filter.ControlCharactersFilter;
+
+import com.xpn.xwiki.internal.store.StoreConfiguration;
+
+import static java.util.Arrays.asList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for testing the {@code Refactoring.Code.RefactoringConfiguration} wiki page.
+ *
+ * @version $Id$
+ * @since 13.2RC1
+ * @since 12.10.6
+ */
+@XWikiSyntax21ComponentList
+@ComponentList({
+    HTMLMacroHTML5Renderer.class,
+    HTML5BlockRenderer.class,
+    HTML5Renderer.class,
+    HTML5RendererFactory.class,
+    DefaultXHTMLLinkRenderer.class,
+    DefaultXHTMLLinkTypeRenderer.class,
+    DefaultXHTMLImageRenderer.class,
+    DefaultXHTMLImageTypeRenderer.class,
+    ControlCharactersFilter.class
+})
+class RefactoringConfigurationTest extends PageTest
+{
+    public static final DocumentReference REFACTORING_CONFIGURATION_REFERENCE =
+        new DocumentReference("xwiki", asList("Refactoring", "Code"), "RefactoringConfiguration");
+
+    public static final String SKIN_PROPERTIES_PATH = "/skins/flamingo/skin.properties";
+
+    @MockComponent
+    private StoreConfiguration storeConfiguration;
```

Comment:
Should probably be move to page test too? 
This is used to mock `isRecycleBinEnabled` in https://github.com/xwiki/xwiki-platform/blob/17f5c241888ea8e986d15f4230f8eec125f5cdd5/xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java#L7036